### PR TITLE
Proxy Service - first e2e test

### PR DESCRIPTION
**Pull request by engelmi**
This PR introduces the first e2e test for the hirte proxy service feature. It also 
- adds classes for the hirte configuration to improve the re-usability (incl. default fixtures)
- adds a hirte container class, which 
   - automatically applies the config to the container and starts the respective hirte services
   - contains (and can be extended by) common functions, e.g. waiting for the hirte systemd service being active
- adds a hirte test class, which
   - takes an arbitrary number of (hirte and hirte-agent) containers and starts/stops them
   - provides a unified test execution routine (incl. gathering logs, cleaning up containers, etc.)
   - prints additional information, e.g. about the used config, to the tmt logs for easier debugging. 
```bash
Starting container hirte-node 'node-foo' with config:
[hirte-agent]
NodeName=node-foo
ManagerHost=10.1.0.37
ManagerPort=8420
ManagerAddress=
HeartbeatInterval=
LogLevel=DEBUG
LogTarget=journald
LogIsQuiet=False
```
- adds `pytest-timeout` so timeouts can be defined for each test individually

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -2,5 +2,5 @@
 # GPL-2.0-or-later
 
 pytest_plugins = [
-    "hirte_tests.common.fixtures"
+    "tests.lib.fixtures"
```
Comment (mwperina) [inline tests/conftest.py:5]: Not 100% persuaded about this change. I've used `hirte_tests` directory/package name to clearly distinguished that this directory is not part of the standard tmt directory structure (like `tests` or `plans`)

File: tests/tests/lib/container.py [hunk 2]
```diff
@@ -62,13 +61,25 @@ def cleanup(self):
...
         return result, output
-    
-    def systemctl_daemon_reload(self) -> Tuple[Optional[int],Union[Iterator[bytes],Any,Tuple[bytes,bytes]]]:
+
+    def systemctl_daemon_reload(self) -> Tuple[Optional[int], Union[Iterator[bytes], Any, Tuple[bytes, bytes]]]:
         return self.exec_run("systemctl daemon-reload")
+
+    def wait_for_hirte(self):
+        result = 1
+        while result != 0:
```
Comment (mwperina) [inline tests/tests/lib/container.py:79]: Indefinite loops are dangerous in tests, we need to have some timeout to fail the tests when it passes.
As an examle please take a look at https://github.com/oVirt/ovirt-system-tests/blob/master/ost_utils/assert_utils.py

File: tests/tests/tier1/hirte-proxy-service/main.fmf [hunk 3]
```diff
@@ -0,0 +1,6 @@
+summary: Tests for the proxy service feature of hirte to resolve dependencies on other nodes
+
+tier: 1
```
Comment (mwperina) [inline tests/tests/tier1/hirte-proxy-service/main.fmf:3]: Is there a reason why not to have basic proxy test in tier0? I think that in tier1 there should be only more advanced or corner case tests, but all important functionality needs to be tests within tier 0

Comment (engelmi) [reply tests/tests/tier1/hirte-proxy-service/main.fmf:3]: Not really. I didn't really think about it, to be honest. It makes sense to group it like you explained. I'll move it to tier 0.

File: tests/tests/lib/container.py (hunk 2 above)
Comment (engelmi) [reply tests/tests/lib/container.py:79]: Definitely, and it is not a nice way to wait for hirte/-agent, but I couldn't think of sth better, unfortunately. 
I added `pytest-timeout` to the list of packages to mark each test individually with annotations - see [the proxy test](https://github.com/containers/hirte/pull/287/files#diff-1ccf81385cf06da6e891f723a10e0982a15d1d844d76a9c098e40aba6b0dc932R13) for example. I didn't add this to the other tests, yet.

File: tests/conftest.py (hunk 1 above)
Comment (engelmi) [reply tests/conftest.py:5]: Because we have the fixtures and other test-related code there, it made sense to locate them rather close to the actual test code. But I understand your point and it makes sense to move the `lib` out of the `test` directory. I'd move it one level up (where it was before) and have `hirte_test/lib`. What do you think? @mwperina

Comment (mwperina) [reply tests/conftest.py:5]: I definitely agree to move outside tmt directory structure. But not sure if it makes sense to have empty package `hirte_test` and then action code in subpackage `lib`, isn't having the code in `hirte_test` enough? But I'm not a python expert so other reviewers might have different suggestions ...

File: tests/tests/lib/container.py (hunk 2 above)
Comment (mwperina) [reply tests/tests/lib/container.py:79]: OK, so then let's add `TODO` here and let's fix in upcoming patches

File: tests/tests/tier1/hirte-proxy-service/main.fmf (hunk 3 above)
Comment (mwperina) [reply tests/tests/tier1/hirte-proxy-service/main.fmf:3]: +1

**Comment by engelmi** [comment]
Closing in favor of #290
